### ISSUE TITLE
Minor issue with semicolon and escaped quote

### ISSUE THREAD

**Issue by przepompownia** [the-mikedavis/tree-sitter-git-config#8]
@the-mikedavis thanks for #4. It fixes almost all cases described there. I found yet one case to fix:
```gitconfig
[alias]
    foo = grep -- \"x\"
    test = "!f() { echo test; }; f"
```
which in Neovim looks like below:
![image](https://github.com/the-mikedavis/tree-sitter-git-config/assets/11404453/4d03d906-c272-4131-ba22-741832687b86)

**Comment by AriSweedler** [comment]
Hi @przepompownia, I think that the issue is not actually in the grammar, but in the use of escape sequences outside of strings. I went ahead and looked for the git docs about gitconfig syntax: https://git-scm.com/docs/git-config#_syntax and read:

> If value needs to contain leading or trailing whitespace characters, it must be enclosed in double quotation marks ("). Inside double quotation marks, double quote (") and backslash (\\) characters must be escaped: use \\" for " and \\\\ for \\.

Then, I wrote some test cases and observed. It looks like putting quotes around the `foo` key made it succeed. And when it was failing, that's when the `test` key got mishighlighted. In my `nvim` instance:

Escape sequences outside of quotes causes an error:
<img width="347" alt="Image" src="https://github.com/user-attachments/assets/05d8ec41-bf6f-43b9-818c-fe96bc8cbcd6" />

Now that the escape sequences are in quotes, everything is good:
<img width="347" alt="Image" src="https://github.com/user-attachments/assets/6c2cba94-96a3-41b5-a080-a533f00ed80d" />

**Comment by przepompownia** [comment]
Hi @AriSweedler thanks for sharing the result of investigation. Good to know although currently I cannot reproduce this behavior. 

![Image](https://github.com/user-attachments/assets/123a3491-be72-4b96-b3cb-44dd09dc92cb)

**Comment by AriSweedler** [comment]
The parser ERRORs out when you use escape sequences outside of strings. You are still doing that! What do you get when you don’t do that?

**Comment by przepompownia** [comment]
Right, the parser error is still present and it affect incremental selection or another range-based things - only highlighting looks better (as on the above screenshot). 

Maybe due to the quoted hint we can close this issue.
